### DISCUSSION TITLE
Capture feature removed?

### DISCUSSION THREAD

**Discussion by FlatMapIO** [alnitak/flutter_soloud#112] [Q&A] [answered]
I noticed that a recent [PR](https://github.com/alnitak/flutter_soloud/pull/106) completely removed the capture feature. I’ve been using this feature in my code to visualize voice input. If the feature has been removed, what alternatives do I have to visualize voice input?

**Comment by themailman05** [comment, marked as answer]
You can fork and keep the capture code :)

**Comment by alnitak** [comment]
Yes sorry. That was an experimental feature that wasn't worth maintaining for the few things it could do.

Maybe in the future, if some other collaborators will join, we will reinsert it.

**Reply by themailman05** [reply]
@alnitak I'm working on a fork with full audio capture and wav writing. Will share soon!

**Comment by FlatMapIO** [comment]
If the audio captured by the external library can be adapted to realtime AudioSource, it might also solve the problem. I don't have experience in this area, so I'm not sure if this is feasible. Looking forward to seeing @themailman05 progress.

**Reply by themailman05** [reply]
I will share soon. It's a bit hacky - essentially I allocate a float buffer in dart and pass a pointer to it in the initialization of the Capture module. In the miniaudio capture callback, I write the frames to that buffer and increment the captured frame count. It works pretty well actually on all platforms I've tested except web.

**Reply by FlatMapIO** [reply]
Looking forward to it! I think I’ll be the first user and tester. 😄

**Comment by alnitak** [comment]
I have published a recorder plugin [here](https://github.com/alnitak/flutter_recorder).
It should work on all platforms web included, but I cannot test on mac/iOS.

It uses miniaudio and can record also skipping silence.

So let's move there for any discussion!
